### PR TITLE
[ARCTIC-920][Optimize] Fix the problem of data inconsistency after minor optimize

### LINKED ISSUE #920: [Bug][Optimize]:If the interval between the major optimize commit and the next minor optimize plan too short will cause data inconsistency
### What happened?

For the arctic keyed table and using the upsert mode, there are two records with the same primary key after the minor optimize.
When we do the minor optimize plan, we get the current snapshot to check whether the base table cache is all in the file cache. Still, we don't force refresh the base table, so maybe we get the old snapshot rather than the latest snapshot if the interval between this minor optimize plan and the last major optimize commit is too short(less than the time arctic table to refresh in cache). If the file cache doesn't have the latest snapshots and files, the check will pass, and we will get the old data files in the base table to do the minor optimize. 
The result is the eq-delete files will apply to the old data files rather than the latest data files. The generated pos-delete files are useless, and the deleted data will lose.

### Affects Versions

0.3.0

### What engines are you seeing the problem on?

_No response_

### How to reproduce

_No response_

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hzluting**
## Why are the changes needed?
fix #920 

## Brief change log

  - *force refresh the base table when checking the file cache in the base table*
  - *remove isRunning in TableOptimizeRuntime and use optimize task size to check whether the table is running*
  - *generate partition running status before getting data files to avoid incorrect plan*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BaseArcticOptimizePlan.java [hunk 1]
```diff
@@ -183,12 +183,11 @@ protected BaseOptimizeTask buildOptimizeTask(@Nullable List<DataTreeNode> source
   }
 
   public boolean baseTableCacheAll() {
-    Snapshot snapshot;
     if (arcticTable.isKeyedTable()) {
-      snapshot = arcticTable.asKeyedTable().baseTable().currentSnapshot();
-      if (snapshot != null && !snapshotIsCached.test(snapshot.snapshotId())) {
+      long snapshotId = UnKeyedTableUtil.getSnapshotId(arcticTable.asKeyedTable().baseTable());
```
Comment (wangtaohz) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BaseArcticOptimizePlan.java:187]: The `baseTable` is refreshed twice in `tableNeedPlan()` and `baseTableCacheAll()`

Comment (hzluting) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BaseArcticOptimizePlan.java:187]: OK, I move it to baseTableCacheAll()

Comment (zhoujinsong) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BaseArcticOptimizePlan.java:187]: How about refreshing both `BaseStore` and `ChangeStore` first in `tableNeedPlan` method,  
This will make the code easier to understand.

Comment (hzluting) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BaseArcticOptimizePlan.java:187]: OK

File: ams/ams-server/src/test/java/com/netease/arctic/ams/server/OptimizeIntegrationTest.java [hunk 2]
```diff
@@ -118,6 +118,9 @@ public static void beforeClass() {
     amsEnvironment.createIcebergCatalog(ICEBERG_CATALOG, ICEBERG_CATALOG_DIR);
     catalog(CATALOG).createDatabase(DATABASE);
     catalog(ICEBERG_CATALOG).createDatabase(DATABASE);
+
+    // wait for ams init service
+    Thread.sleep(1000);
```
Comment (wangtaohz) [inline ams/ams-server/src/test/java/com/netease/arctic/ams/server/OptimizeIntegrationTest.java:123]: It's not guaranteed to finish initiation after waiting for 1s.

**Review by wangtaohz — APPROVED** [review]

**Comment by wangtaohz** [comment]
LGTM

**Review by zhoujinsong — APPROVED** [review]
LGTM.
